### ISSUE TITLE
Great project! But I have some problems;

### ISSUE THREAD

**Issue by Byronnar** [HuCaoFighting/Swin-Unet#17]
Would you please tell me the best performence that you trained? 
I have trained the model ith default parameters in Synapse datasets, but the performence is not good, as below:
```
Testing performance in best val model: mean_dice : 0.760341 mean_hd95 : 29.199245
```
It is really diffirent with the paper, how should I do? 
Thank you!

**Comment by HuCaoFighting** [comment]
Do you train your model based on this git?

**Comment by Byronnar** [comment]
> Do you train your model based on this git?

Yes， and the datasets from Tranunet.

**Comment by HuCaoFighting** [comment]
OK, we will check it carefully. Thanks for your feedback.

**Comment by Byronnar** [comment]
> OK, we will check it carefully. Thanks for your feedback.

Thank you! my params as below:
```
parser = argparse.ArgumentParser()
parser.add_argument('--root_path', type=str,
                    default='../datasets/Synapse', help='root dir for data')
parser.add_argument('--dataset', type=str,
                    default='Synapse', help='experiment_name')
parser.add_argument('--list_dir', type=str,
                    default='./lists/lists_Synapse', help='list dir')
parser.add_argument('--num_classes', type=int,
                    default=9, help='output channel of network')
parser.add_argument('--output_dir', type=str, default='./model_out', help='output dir')
parser.add_argument('--max_iterations', type=int,
                    default=30000, help='maximum epoch number to train')    # 30000
parser.add_argument('--max_epochs', type=int,
                    default=150, help='maximum epoch number to train')  # 150
parser.add_argument('--batch_size', type=int,
                    default=24, help='batch_size per gpu')
parser.add_argument('--n_gpu', type=int, default=1, help='total gpu')
parser.add_argument('--deterministic', type=int,  default=1,
                    help='whether use deterministic training')
parser.add_argument('--base_lr', type=float,  default=0.01,
                    help='segmentation network learning rate')
parser.add_argument('--img_size', type=int,
                    default=224, help='input patch size of network input')
parser.add_argument('--seed', type=int,
                    default=1234, help='random seed')
parser.add_argument('--cfg', type=str, required=False, default='configs/swin_tiny_patch4_window7_224_lite.yaml', metavar="FILE", help='path to config file', )
parser.add_argument(
        "--opts",
        help="Modify config options by adding 'KEY VALUE' pairs. ",
        default=None,
        nargs='+',
    )
parser.add_argument('--zip', action='store_true', help='use zipped dataset instead of folder dataset')
parser.add_argument('--cache-mode', type=str, default='part', choices=['no', 'full', 'part'],
                    help='no: no cache, '
                            'full: cache all data, '
                            'part: sharding the dataset into nonoverlapping pieces and only cache one piece')
parser.add_argument('--resume', help='resume from checkpoint')
parser.add_argument('--accumulation-steps', type=int, help="gradient accumulation steps")
parser.add_argument('--use-checkpoint', action='store_true',
                    help="whether to use gradient checkpointing to save memory")
parser.add_argument('--amp-opt-level', type=str, default='O1', choices=['O0', 'O1', 'O2'],
                    help='mixed precision opt level, if O0, no amp is used')
parser.add_argument('--tag', help='tag of experiment')
parser.add_argument('--eval', action='store_true', help='Perform evaluation only')
parser.add_argument('--throughput', action='store_true', help='Test throughput only')

args = parser.parse_args()
if args.dataset == "Synapse":
    args.root_path = os.path.join(args.root_path, "train_npz")
config = get_config(args)
```
And the ddataset from TransUnet procecssed Synapse;

**Comment by yangjingyi** [comment]
Hi,

Thanks for public code of this paper!
I have a question about the test result. I use the default parameter of code to train and test. And I get the result as below:
Testing performance in best val model: mean_dice : 0.784067 mean_hd95 : 21.433199
A little different with results listed in the paper. 
I found that you use DEPTH 2222 in your yaml file. But according to the Fig1 in paper. It seems that you use DEPTH 2221. What block number you use to get the best result you list in the paper?
Thanks in advance!

**Comment by zhouyi-git** [comment]
> > 你是否基于这个 git 训练你的模型？
> 
> 是的，还有来自 Tranunet 的数据集。

Can you give a parameter of 0.79

**Comment by zhouyi-git** [comment]
> 你好，
> 
> 感谢本文的公开代码！
> 我对测试结果有疑问。我使用代码的默认参数来训练和测试。我得到的结果如下：
> 在最佳 val 模型中测试性能： mean_dice : 0.784067 mean_hd95 : 21.433199
> 与论文中列出的结果略有不同。
> 我发现您在 yaml 文件中使用了 DEPTH 2222。但是根据论文中的Fig1。您似乎使用 DEPTH 2221。您使用什么块号来获得您在论文中列出的最佳结果？
> 提前致谢！

Has your problem been solved

**Comment by zhouyi-git** [comment]
> > 你好，
> > 感谢本文的公开代码！
> > 我对测试结果有疑问。我使用代码的默认参数来训练和测试。我得到的结果如下：
> > 在最佳 val 模型中测试性能： mean_dice : 0.784067 mean_hd95 : 21.433199
> > 与论文中的结果略有不同。
> > 我发现您在YAML文件中使用了深度2222但是根据论文中的FIG1。您似乎使用深度2221您使用什么块号来获得您在论文中列出的最佳结果？
> > 提前致谢！
> 
> 你的问题解决了吗

Can you give your parameters

**Comment by yangjingyi** [comment]
> > > 你好，
> > > 感谢本文的公开代码！
> > > 我对测试结果有疑问。我使用代码的默认参数来训练和测试。我得到的结果如下：
> > > 在最佳 val 模型中测试性能： mean_dice : 0.784067 mean_hd95 : 21.433199
> > > 与论文中的结果略有不同。
> > > 我发现您在YAML文件中使用了深度2222但是根据论文中的FIG1。您似乎使用深度2221您使用什么块号来获得您在论文中列出的最佳结果？
> > > 提前致谢！
> > 
> > 
> > 你的问题解决了吗
> 
> Can you give your parameters

NO, I don't solve this problem.
Got this result mean_dice : 0.784067 mean_hd95 : 21.433199 with this set of parameters:
MODEL:
  TYPE: swin
  NAME: swin_tiny_patch4_window7_224
  DROP_PATH_RATE: 0.2
  PRETRAIN_CKPT: "./pretrained_ckpt/swin_tiny_patch4_window7_224.pth"
  SWIN:
    FINAL_UPSAMPLE: "expand_first"
    EMBED_DIM: 96
    DEPTHS: [ 2, 2, 2, 1 ]
    DECODER_DEPTHS: [ 2, 2, 2, 1]
    NUM_HEADS: [ 3, 6, 12, 24 ]
    WINDOW_SIZE: 7

**Comment by wddwzwhhxx** [comment]
> 

Mean class 1 mean_dice 0.869020 mean_hd95 0.000000
Mean class 2 mean_dice 0.790491 mean_hd95 0.000000
Mean class 3 mean_dice 0.811839 mean_hd95 0.000000
Mean class 4 mean_dice 0.659608 mean_hd95 0.000000
Mean class 5 mean_dice 0.931941 mean_hd95 0.000000
Mean class 6 mean_dice 0.697862 mean_hd95 0.000000
Mean class 7 mean_dice 0.847112 mean_hd95 0.000000
Mean class 8 mean_dice 0.558213 mean_hd95 0.000000
Testing performance in best val model: mean_dice : 0.770761 mean_hd95 : 0.000000

我甚至达不到你的0.784的效果，请问你有修改什么参数或者代码吗

**Comment by wddwzwhhxx** [comment]
> > > > 你好，
> > > > 感谢本文的公开代码！
> > > > 我对测试结果有疑问。我使用代码的默认参数来训练和测试。我得到的结果如下：
> > > > 在最佳 val 模型中测试性能： mean_dice : 0.784067 mean_hd95 : 21.433199
> > > > 与论文中的结果略有不同。
> > > > 我发现您在YAML文件中使用了深度2222但是根据论文中的FIG1。您似乎使用深度2221您使用什么块号来获得您在论文中列出的最佳结果？
> > > > 提前致谢！
> > > 
> > > 
> > > 你的问题解决了吗
> > 
> > 
> > Can you give your parameters
> 
> NO, I don't solve this problem. Got this result mean_dice : 0.784067 mean_hd95 : 21.433199 with this set of parameters: MODEL: TYPE: swin NAME: swin_tiny_patch4_window7_224 DROP_PATH_RATE: 0.2 PRETRAIN_CKPT: "./pretrained_ckpt/swin_tiny_patch4_window7_224.pth" SWIN: FINAL_UPSAMPLE: "expand_first" EMBED_DIM: 96 DEPTHS: [ 2, 2, 2, 1 ] DECODER_DEPTHS: [ 2, 2, 2, 1] NUM_HEADS: [ 3, 6, 12, 24 ] WINDOW_SIZE: 7

Mean class 1 mean_dice 0.869020 mean_hd95 0.000000
Mean class 2 mean_dice 0.790491 mean_hd95 0.000000
Mean class 3 mean_dice 0.811839 mean_hd95 0.000000
Mean class 4 mean_dice 0.659608 mean_hd95 0.000000
Mean class 5 mean_dice 0.931941 mean_hd95 0.000000
Mean class 6 mean_dice 0.697862 mean_hd95 0.000000
Mean class 7 mean_dice 0.847112 mean_hd95 0.000000
Mean class 8 mean_dice 0.558213 mean_hd95 0.000000
Testing performance in best val model: mean_dice : 0.770761 mean_hd95 : 0.000000

我甚至达不到你的0.784的效果，请问你有修改什么参数或者代码吗

**Comment by caoshiwen1** [comment]
> different with results listed in th

你好，可以分享一下数据集吗，联系作者未果，谢谢1816383535@qq.com

**Comment by yangjingyi** [comment]
> 

No. I haven't modified any parameter. You could check your pretrained model and dataset.

**Comment by yangjingyi** [comment]
> > > > > 你好，
> > > > > 感谢本文的公开代码！
> > > > > 我对测试结果有疑问。我使用代码的默认参数来训练和测试。我得到的结果如下：
> > > > > 在最佳 val 模型中测试性能： mean_dice : 0.784067 mean_hd95 : 21.433199
> > > > > 与论文中的结果略有不同。
> > > > > 我发现您在YAML文件中使用了深度2222但是根据论文中的FIG1。您似乎使用深度2221您使用什么块号来获得您在论文中列出的最佳结果？
> > > > > 提前致谢！
> > > > 
> > > > 
> > > > 你的问题解决了吗
> > > 
> > > 
> > > Can you give your parameters
> > 
> > 
> > NO, I don't solve this problem. Got this result mean_dice : 0.784067 mean_hd95 : 21.433199 with this set of parameters: MODEL: TYPE: swin NAME: swin_tiny_patch4_window7_224 DROP_PATH_RATE: 0.2 PRETRAIN_CKPT: "./pretrained_ckpt/swin_tiny_patch4_window7_224.pth" SWIN: FINAL_UPSAMPLE: "expand_first" EMBED_DIM: 96 DEPTHS: [ 2, 2, 2, 1 ] DECODER_DEPTHS: [ 2, 2, 2, 1] NUM_HEADS: [ 3, 6, 12, 24 ] WINDOW_SIZE: 7
> 
> Mean class 1 mean_dice 0.869020 mean_hd95 0.000000 Mean class 2 mean_dice 0.790491 mean_hd95 0.000000 Mean class 3 mean_dice 0.811839 mean_hd95 0.000000 Mean class 4 mean_dice 0.659608 mean_hd95 0.000000 Mean class 5 mean_dice 0.931941 mean_hd95 0.000000 Mean class 6 mean_dice 0.697862 mean_hd95 0.000000 Mean class 7 mean_dice 0.847112 mean_hd95 0.000000 Mean class 8 mean_dice 0.558213 mean_hd95 0.000000 Testing performance in best val model: mean_dice : 0.770761 mean_hd95 : 0.000000
> 
> 我甚至达不到你的0.784的效果，请问你有修改什么参数或者代码吗

No. I haven't modified any parameter. You could check your pretrained model and dataset.

**Comment by ninedow** [comment]
> > different with results listed in th
> 
> 你好，可以分享一下数据集吗，联系作者未果，谢谢[1816383535@qq.com](mailto:1816383535@qq.com)

你好，请问你收到处理好的数据集了吗？可以的话能分享一下吗，我也联系作者未果，非常感谢！(email:1002203661@qq.com)

**Comment by wangru1026** [comment]
我是用原始数据集自己预处理得到的数据集，但是我的num class 设置的是14，我看的Synapse官网显示的分类也是13类（加背景是14），想问一下大家的num class默认值9就能跑通？都没有调过吗？

**Comment by EvgeniaChroni** [comment]
I have a similar issue I use the data provided from TransUnet repo for Synopsis dataset. 
However I cannot reproduce the results for the test set. 

What I get as results are: 

Mean class 1 mean_dice 0.000000 mean_hd95 135.274750
Mean class 2 mean_dice 0.000000 mean_hd95 99.498457
Mean class 3 mean_dice 0.916550 mean_hd95 6.559290
Mean class 4 mean_dice 0.166667 mean_hd95 92.171695
Mean class 5 mean_dice 0.000157 mean_hd95 165.124671
Mean class 6 mean_dice 0.000389 mean_hd95 123.926535
Mean class 7 mean_dice 0.036210 mean_hd95 134.851290
Mean class 8 mean_dice 0.000079 mean_hd95 102.834936
Testing performance in best val model: mean_dice : 0.140006 mean_hd95 : 107.530203

I made no change in the code. 
Could you please let us know how can we fix it? Or upload the data that you used? I cannot be sure if the problem is because of the data or your code.

**Comment by Flame-circle** [comment]
> 我有一个类似的问题，我使用从TransUnet存储库提供的数据用于概要数据集。但是我无法重现测试集的结果。
> 
> 我得到的结果是：
> 
> 平均类 1 mean_dice 0.000000 mean_hd95 135.274750 平均类 2 mean_dice 0.000000 mean_hd95 99.498457 平均值类 3 mean_dice 0.916550 mean_hd95 6.559290 平均类 4 mean_dice 0.166667 mean_hd95 92.171695 均值类 5 mean_dice 0.000157 mean_hd95 165.124671 均值类 6 mean_dice 0.000389 mean_hd95 123.926535 平均类 7 mean_dice 0.036210 mean_hd95 134.851290 平均类 8 mean_dice 0.000079 mean_hd95 102.834936最佳值模型中的测试性能： mean_dice ： 0.140006 mean_hd95 ： 107.530203
> 
> 我没有对代码进行任何更改。你能告诉我们如何解决它吗？或者上传您使用的数据？我不确定问题是由于数据还是您的代码。

我使用了一个遥感图像数据集，也没有修改模型参数，得到了和你类似的结果，预测输出可视化后发现分割边缘是锯齿状。请问你的问题解决了吗？如果解决了麻烦告诉我是如何解决的，谢谢

> 我有一个类似的问题，我使用从TransUnet存储库提供的数据用于概要数据集。但是我无法重现测试集的结果。
> 
> 我得到的结果是：
> 
> 平均类 1 mean_dice 0.000000 mean_hd95 135.274750 平均类 2 mean_dice 0.000000 mean_hd95 99.498457 平均值类 3 mean_dice 0.916550 mean_hd95 6.559290 平均类 4 mean_dice 0.166667 mean_hd95 92.171695 均值类 5 mean_dice 0.000157 mean_hd95 165.124671 均值类 6 mean_dice 0.000389 mean_hd95 123.926535 平均类 7 mean_dice 0.036210 mean_hd95 134.851290 平均类 8 mean_dice 0.000079 mean_hd95 102.834936最佳值模型中的测试性能： mean_dice ： 0.140006 mean_hd95 ： 107.530203
> 
> 我没有对代码进行任何更改。你能告诉我们如何解决它吗？或者上传您使用的数据？我不确定问题是由于数据还是您的代码。

我使用了一个遥感图像数据集，也没有修改模型参数，得到了和你类似的结果，预测输出可视化后发现分割边缘是锯齿状。请问你的问题解决了吗？如果解决了麻烦告诉我是如何解决的，谢谢

**Comment by YumnaMemon** [comment]
> Would you please tell me the best performence that you trained? I have trained the model ith default parameters in Synapse datasets, but the performence is not good, as below:
> 
> ```
> Testing performance in best val model: mean_dice : 0.760341 mean_hd95 : 29.199245
> ```
> 
> It is really diffirent with the paper, how should I do? Thank you!

can you add me on wechat: yahm30

**Comment by syhkqa** [comment]
我也没有0.78，你们解决了吗
+-
